src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -735,6 +736,11 @@ MultiClientCopyData(int32 connectionId, int32 fileDescriptor)
 		int appended = -1;
 		errno = 0;
 
+		if (SetResultLimit)
+		{
+			currentIntermediateResult += receiveLength;
```

Comment:
`MultiClientCopyData ` is used for things which have nothing do with `currentIntermediateResult`

It'd be nicer to add a `int* returnBytesReceived` parameter and do something like:
```
if (returnBytesReceived != NULL)
{
    *returnBytesReceived = receiveLength;
}
```